src/App.jsx

```diff
@@ -14,45 +19,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function sortGoodsBy(goodList, sortBy, isNeedReverse) {
+  const goodListCopy = [...goodList];
+
+  goodListCopy.sort((good1, good2) => {
+    switch (sortBy) {
+      case ('alpabetic'):
+        return good1.localeCompare(good2);
+
+      case ('length'):
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isNeedReverse) {
+    goodListCopy.reverse();
+  }
+
+  return goodListCopy;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isChanged, setIsChanged] = useState(false);
+  const goods = sortGoodsBy(goodsFromServer, sortBy, isReversed);
+
+  const resetButton = (
+    <button
+      type="button"
+      className="button is-danger is-light"
+      onClick={() => {
+        setSortBy('');
+        setIsReversed(false);
+        setIsChanged(false);
+      }}
+    >
+      Reset
+    </button>
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortBy !== SORT_BY_ALPHABETIC,
+          })}
+          onClick={() => {
+            setSortBy(SORT_BY_ALPHABETIC);
+            setIsChanged(true);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortBy !== SORT_BY_LENGTH,
+          })}
+          onClick={() => {
+            setSortBy(SORT_BY_LENGTH);
+            setIsChanged(true);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(!isReversed);
```

Comment:
toggling is depending on previous state
```suggestion
          setIsReversed(isReversed => !isReversed);
```